### PR TITLE
C# Improve default argument value extraction

### PR DESCRIPTION
**Pull request by tamasvajk**
[Differences job](https://jenkins.internal.semmle.com/job/Changes/job/CSharp-Differences/990/)

### CODE REVIEW

File: csharp/ql/test/library-tests/csharp7/LocalTaintFlow.ql [hunk 1]
```diff
@@ -2,5 +2,7 @@ import csharp
 import semmle.code.csharp.dataflow.TaintTracking
 
 from DataFlow::Node pred, DataFlow::Node succ
-where TaintTracking::localTaintStep(pred, succ)
+where
+  TaintTracking::localTaintStep(pred, succ) and
+  not pred.asExpr().fromLibrary()
```
Comment (hvitved) [inline csharp/ql/test/library-tests/csharp7/LocalTaintFlow.ql:7]: Why not `pred.asExpr().fromSource()`

**Review by hvitved — COMMENTED** [review]
Looks great. A few trivial comments.

File: csharp/ql/test/library-tests/csharp7/LocalTaintFlow.ql (hunk 1 above)
Comment (tamasvajk) [reply csharp/ql/test/library-tests/csharp7/LocalTaintFlow.ql:7]: I tried that, but that would remove SSA definitions, nodes for implicit `this` expressions, and maybe more.

Comment (hvitved) [reply csharp/ql/test/library-tests/csharp7/LocalTaintFlow.ql:7]: 👍

**Review by hvitved — APPROVED** [review]

**Pull request merged**